File: 392. Is Subsequence/isSubsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const isSubsequence = (s, t) => {
  if (!s) {
    return true;
  }

  let i = 0;
  for (let j = 0; j < t.length; j++) {
    if (t[j] === s[i]) {
      i++;
      if (i === s.length) {
        return true;
      }
    }
  }

  return false;
};


const s = "b";
const t = "abc";
console.log("isSubsequence: ",  isSubsequence(s, t))

File: 392. Is Subsequence/isSubsequence.py
class Solution:
  def isSubsequence(self, s: str, t: str) -> bool:
    if not s:
      return True
    
    i = 0
    for j in range(len(t)):
      if t[j] == s[i]:
        i += 1
        if i == len(s):
          return True

    return False
  

s = "b"
t = "abc"
solution = Solution()
print("isSubsequence: ",  solution.isSubsequence(s, t))
